src/ui-test/camelk_java_debug_test.ts

```diff
@@ -0,0 +1,201 @@
+	await startMode.selectQuickPick('Basic');
+
+	await VSBrowser.instance.driver.sleep(1000);
+}
+
+async function createIntegration(fileName: string) {
+	const workbench = new Workbench();
+	await workbench.executeCommand('Create a new Apache Camel K Integration file');
+	const languageInput = await InputBox.create();
+	await languageInput.selectQuickPick('Java');
+	const WORKSPACE_FOLDERInput = await InputBox.create();
+	await WORKSPACE_FOLDERInput.selectQuickPick(0);
+	const nameInput = await InputBox.create();
+	await nameInput.setText(fileName);
+	await nameInput.confirm();
+
+	const editorView = new EditorView();
+	await VSBrowser.instance.driver.wait(async() => {
+		try {
+			return await editorView.openEditor(fileName + '.java') !== undefined;
+		} catch {
+			return false;
+		}
+	});
+	console.log('integration created');
+	return workbench;
+}
+
+async function modifyCurrentFileToBeInvalid() {
+	const textEditor : TextEditor = new TextEditor();
+	await textEditor.setTextAtLine(14, ";");
+	await textEditor.save()
+}
+
+async function addBreakpointToFile() {
+	const textEditor : TextEditor = new TextEditor();
+	await textEditor.toggleBreakpoint(13);
+	await textEditor.save();
+}
+
+async function removeIntegration(integrationLabel: string) {
+	const section = await getIntegrationSectionFromSideView(integrationLabel);
+	const item = await section.findItem(integrationLabel) as ViewItem;
+	const menu = await item.openContextMenu();
+	const removeItem = await menu.getItem(REMOVE_INTEGRATION_LABEL);
+	await removeItem?.click();
+}
+
+async function prepareTempWorkspaceForTests(workspaceFolder: string) {
+	await new EditorView().closeAllEditors();
+	await prepareEmptyTestFolder(workspaceFolder);
+	await VSBrowser.instance.openResources(workspaceFolder);
+}
+
+async function findIntegrationOnSideBar(integrationLabel: string) {
+	const section = await getIntegrationSectionFromSideView(integrationLabel);
+	return await section.findItem(integrationLabel) as ViewItem;
+}
+
+async function getIntegrationSectionFromSideView(integrationLabel: string) {
+	const section = await new SideBarView().getContent().getSection('Apache Camel K Integrations') as CustomTreeSection;
+	await section.expand();
+	await VSBrowser.instance.driver.sleep(5000);
+	return section;
+}
+
+async function stopAndExitDebug() {
+	await stopDebugging();
+	await getIntoExplorerView();
+}
+
+async function stopDebugging() {
+	await (await DebugToolbar.create()).stop();
+}
+
+async function getIntoExplorerView() {
+	const control = await new ActivityBar().getViewControl('Explorer');
+	await control?.openView();
+}
+
+async function getContextMenuOf(integrationLabel: string) {
+	const item = await findIntegrationOnSideBar(integrationLabel);
+	return await item.openContextMenu()
+}
+
+async function startDebuggerOn(integrationLabel: string) {
+	const menu = await getContextMenuOf(integrationLabel);
+
+	await menu.getItem(START_DEBUG_LABEL).then(item => item?.click());
```

Comment:
I would recommend to not mix the 2 notations of `await` and `then`.
Having await everywhere allows to be more consistent.